### PR TITLE
[libc++] Address post-commit comments for __scope_guard

### LINKED ISSUE #116204: [libc++] __scope_guard's constructor should be explicit
This seems to have been missed during the review.

### PR DESCRIPTION
**Pull request by philnik777**
Fixes #116204

### CODE REVIEW

File: libcxx/include/__utility/scope_guard.h [hunk 1]
```diff
@@ -26,26 +26,23 @@ _LIBCPP_BEGIN_NAMESPACE_STD
...
-// C++17 has mandatory RVO, so we don't need the move constructor anymore to make __make_scope_guard work.
+// C++14 doesn't have mandatory RVO, so we have to provide a declaration even though no compiler will ever generate
+// a call to the move constructor.
 #if _LIBCPP_STD_VER <= 14
-  __scope_guard(__scope_guard&& __other) : __func_(__other.__func_) {
-    _LIBCPP_ASSERT_INTERNAL(!__other.__moved_from_, "Cannot move twice from __scope_guard");
-    __other.__moved_from_ = true;
-  }
+private:
+  __scope_guard(__scope_guard&&);
```
Comment (philnik777) [inline libcxx/include/__utility/scope_guard.h:42]: I'm not sure whether this approach is better than having dead code. @ldionne thoughts?

Comment (ldionne) [reply libcxx/include/__utility/scope_guard.h:42]: Oh, I like this a lot better than the previous approach. There's not much harm in providing this private declaration.

**Review by ldionne — APPROVED** [review]

File: libcxx/include/__utility/scope_guard.h (hunk 1 above)
Comment (philnik777) [reply libcxx/include/__utility/scope_guard.h:42]: Uh... this actually has to be public. Sometimes my brain just turns off. Does that change your thoughts on this?

Comment (ldionne) [reply libcxx/include/__utility/scope_guard.h:42]: It doesn't, but thanks for checking.

**Pull request merged**